### DISCUSSION TITLE
Search configuration

### DISCUSSION THREAD

**Discussion by lebedevilya** [avo-hq/avo#379] [Help] [answered]
Hello friends, just got my first project set up with Avo, everything seems very cool. I got a question about search function. Documentation clearly shows index view with search bar and action buttons:
![image](https://user-images.githubusercontent.com/6682730/116818380-adc6ae80-ab73-11eb-9a55-a6553286ccfa.png)
But my all of my index views are missing these. I found seach setup guide using @search variable for version 0.4.x but seems like this entry is missing for version 1.0. Could you share some information about setting up search in version 1.0?

One more question is about customization of default avo actions. For example, i needed to run some custom code in create action and i managed to do that like this:
![image](https://user-images.githubusercontent.com/6682730/116818448-1dd53480-ab74-11eb-8f82-6cccf7457fae.png)
It was not so obivous at first, but it works. I didnt find any guide about customizing these default actions, do you have some information to share? Like which variables are available inside of these actions, which actions are available by default, e.t.c.
Thank you!

UPD: just checked demo application in your repo and seems like Search is not working in your demo too

**Comment by adrianthedev** [comment]
Hi Ilya,

This is Adrian, the creator of Avo. 
Search was not such a pleasant experience in 0.4, so we went ahead and skipped the implementation for v1.0. 
We are working on bringing back the search feature. There’s even a PR up with 90% of the work done. I need to give it a once-over to make sure all is good and add tests for it. So expect a release for search this week. You can continue to develop your app with Avo because the update process is painless.

Regarding the controller action update, you did well. You can also use [model callbacks](https://guides.rubyonrails.org/active_record_callbacks.html#creating-an-object) to achieve the same result. 
We haven’t made a lot of docs for how the internals work yet. Until then, You can check the source code for the `BaseController`.

Please let me know if I can help some more.

**Comment by adrianthedev** [comment, marked as answer]
The search is not implemented in the demo app because it runs on v 1.x and, the search feature is not available in v 1.x yet. It will be released this week.

**Reply by lebedevilya** [reply]
Cool, thank you, I'll be waiting for it

**Comment by adrianthedev** [comment]
Coming back to this. We released search last week.
You can find more details about configuration on the [docs page](https://docs.avohq.io/1.0/search.html#search).

**Reply by bart-oz** [reply]
Hi @adrianthedev -> new search is fantastic, especially global search feature, however what can I do to configure it for my own purpose such as translating placeholder, and "cancel" button. Can't see it in locales .yml file. Thank's in advance for help.

**Reply by mihaimdm22** [reply]
Hi,

We implemented the translation of the placeholder and cancel button in version 1.13.2, but it seems that Algolia autocomplete package doesn't forward that cancel button translation. It might be a bug on their side.

Thanks!
